### PR TITLE
Fix Undo/Redo bug

### LINKED ISSUE #203: Undoing consecutive commands on different lists results in wrong list view
If we do:
`delete 1 i/ int`
`delete 2 i/ res`
`undo`
`undo`
We will end up in list view of resume. Expected: list view of int.

### PR DESCRIPTION
**Pull request by wardetu**

### CODE REVIEW

**Comment by wardetu** [comment]
Main changes are the ones begin with `fix`.

The others were my struggle to fix the writing bit/ line-ending (CRLF vs LF) bug.

**Review by helloImHai — APPROVED** [review]
Nice one bro 🥇

File: src/main/java/seedu/address/model/ResumeBook.java [hunk 1]
```diff
@@ -225,7 +226,14 @@ public void deleteInternship(Internship key) {
         internships.remove(key);
         for (Item item : resumes) {
             Resume resume = (Resume) item;
-            resume.getInternshipIds().remove(Integer.valueOf(id));
+            Resume newresume = new Resume(resume.getName(), resume.getId(), resume.getTags());
```
Comment (chrisjwelly) [inline src/main/java/seedu/address/model/ResumeBook.java]: This should be camel case: `newResume`. 
```suggestion
            Resume newResume = new Resume(resume.getName(), resume.getId(), resume.getTags());
```

File: src/main/java/seedu/address/model/ResumeBook.java [hunk 2]
```diff
@@ -225,7 +226,14 @@ public void deleteInternship(Internship key) {
...
+            List<Integer> newInternshipList = new ArrayList<>();
+            newInternshipList.addAll(resume.getInternshipIds());
+            newresume.setInternshipIds(newInternshipList);
+            newresume.setProjectIds(resume.getProjectIds());
+            newresume.setSkillIds(resume.getSkillIds());
+            newresume.getInternshipIds().remove(Integer.valueOf(id));
```
Comment (chrisjwelly) [inline src/main/java/seedu/address/model/ResumeBook.java]: Adding white space for better readability
```suggestion

            newResume.setInternshipIds(newInternshipList);
            newResume.setProjectIds(resume.getProjectIds());
            newResume.setSkillIds(resume.getSkillIds());
            newResume.getInternshipIds().remove(Integer.valueOf(id));
            
```

File: src/main/java/seedu/address/model/ResumeBook.java [hunk 3]
```diff
@@ -291,7 +299,14 @@ public void deleteProject(Project key) {
...
             Resume resume = (Resume) item;
-            resume.getProjectIds().remove(Integer.valueOf(id));
+            Resume newresume = new Resume(resume.getName(), resume.getId(), resume.getTags());
+            newresume.setInternshipIds(resume.getInternshipIds());
+            List<Integer> newProjectList = new ArrayList<>();
+            newProjectList.addAll(resume.getProjectIds());
+            newresume.setProjectIds(newProjectList);
+            newresume.setSkillIds(resume.getSkillIds());
+            newresume.getProjectIds().remove(Integer.valueOf(id));
+            setResume(resume, newresume);
```
Comment (chrisjwelly) [inline src/main/java/seedu/address/model/ResumeBook.java]: Making it consistent with internship:
```suggestion
            Resume newResume = new Resume(resume.getName(), resume.getId(), resume.getTags());
            List<Integer> newProjectList = new ArrayList<>();
            newProjectList.addAll(resume.getProjectIds());
            
            newResume.setInternshipIds(resume.getInternshipIds());
            newResume.setProjectIds(newProjectList);
            newResume.setSkillIds(resume.getSkillIds());
            newResume.getProjectIds().remove(Integer.valueOf(id));
            
            setResume(resume, newResume);
```

File: src/main/java/seedu/address/model/ResumeBook.java [hunk 4]
```diff
@@ -225,7 +226,14 @@ public void deleteInternship(Internship key) {
...
             Resume resume = (Resume) item;
-            resume.getInternshipIds().remove(Integer.valueOf(id));
+            Resume newresume = new Resume(resume.getName(), resume.getId(), resume.getTags());
+            List<Integer> newInternshipList = new ArrayList<>();
+            newInternshipList.addAll(resume.getInternshipIds());
+            newresume.setInternshipIds(newInternshipList);
+            newresume.setProjectIds(resume.getProjectIds());
+            newresume.setSkillIds(resume.getSkillIds());
+            newresume.getInternshipIds().remove(Integer.valueOf(id));
+            setResume(resume, newresume);
```
Comment (chrisjwelly) [inline src/main/java/seedu/address/model/ResumeBook.java]: Oops forgot to batch this one
```suggestion
            setResume(resume, newResume);
```

File: src/main/java/seedu/address/model/ResumeBook.java [hunk 5]
```diff
@@ -357,7 +372,14 @@ public void deleteSkill(Skill key) {
...
             Resume resume = (Resume) item;
-            resume.getSkillIds().remove(Integer.valueOf(id));
+            Resume newresume = new Resume(resume.getName(), resume.getId(), resume.getTags());
+            newresume.setInternshipIds(resume.getInternshipIds());
+            newresume.setProjectIds(resume.getProjectIds());
+            List<Integer> newSkillList = new ArrayList<>();
+            newSkillList.addAll(resume.getSkillIds());
+            newresume.setSkillIds(newSkillList);
+            newresume.getSkillIds().remove(Integer.valueOf(id));
+            setResume(resume, newresume);
```
Comment (chrisjwelly) [inline src/main/java/seedu/address/model/ResumeBook.java]: Making it consistent with internship
```suggestion
            Resume newResume = new Resume(resume.getName(), resume.getId(), resume.getTags());
            List<Integer> newSkillList = new ArrayList<>();
            newSkillList.addAll(resume.getSkillIds());
            
            newResume.setInternshipIds(resume.getInternshipIds());
            newResume.setProjectIds(resume.getProjectIds());
            newResume.setSkillIds(newSkillList);
            newResume.getSkillIds().remove(Integer.valueOf(id));
            
            setResume(resume, newResume);
```

**Review by chrisjwelly — CHANGES_REQUESTED** [review]
Functionality looks okay, but I suggested some improvements with regards to code style. Please check that I have done it correctly and it won't break anything @wardetu

**Review by chrisjwelly — APPROVED** [review]
LGTM! :tada:
(after trailing whitespace merges)

**Pull request merged**